crates/red_knot_python_semantic/src/types.rs

```diff
@@ -185,6 +185,10 @@ pub enum Type<'db> {
     StringLiteral(StringLiteralType<'db>),
     /// A bytes literal
     BytesLiteral(BytesLiteralType<'db>),
+    /// A literal string, like `StringLiteral`, but *without* the value.
+    ///
+    /// Useful for e.g. `1000000 * "hello"`.
```

Comment:
Suggested edit to the doc comment:

```suggestion
    /// A string known to originate only from literal values, but whose value is not known (unlike StringLiteral above).
```

Also suggest placing this next to StringLiteral.